tests/test_buddycheck.py

```diff
@@ -0,0 +1,162 @@
+import unittest
+from unittest.mock import patch
+import pandas as pd
+import tempfile
+import os
+from cautiousrobot import BuddyCheck
+from cautiousrobot.utils import process_csv
+
+class TestBuddyCheck(unittest.TestCase):
+    def setUp(self):
+        self.buddy_check = BuddyCheck()
+        self.buddy_check_filename = BuddyCheck(buddy_id='filename')
+
+        self.img_source_file = tempfile.NamedTemporaryFile(delete=False, mode='w')
+        self.checksum_source_file = tempfile.NamedTemporaryFile(delete=False, mode='w')
+
+        self.img_source_file.write("""filename,checksum
+image1.jpg,abc123
+image2.jpg,def456
+image3.jpg,ghi789
+""")
+        self.img_source_file.close()
+
+        self.checksum_source_file.write("""filename,md5
+image1.jpg,abc123
+image2.jpg,def456
+image3.jpg,ghi789
+""")
+        self.checksum_source_file.close()
+        
+
+    def tearDown(self):
+        os.remove(self.img_source_file.name)
+        os.remove(self.checksum_source_file.name)
+
+    def test_initialization(self):
+        self.assertEqual(self.buddy_check.buddy_id, None)
+        self.assertEqual(self.buddy_check.buddy_col, 'md5')
+
+    def test_merge_on_checksum(self):
+        source_df = pd.read_csv(self.img_source_file.name)
+        checksum_df = pd.read_csv(self.checksum_source_file.name).drop(columns=['filename'])
```

Comment:
```suggestion
        checksum_df = pd.read_csv(self.checksum_source_file.name)
```
It should work regardless of whether the DataFrame has a "filename" column.